### PR TITLE
Modify rule S115: Extend LaYC content

### PR DESCRIPTION
**Pull request by anton-haubner-sonarsource**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S115/common/why-is-this-an-issue.adoc [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+program after initialization.
+Oftentimes, constants are used in multiple locations across different
+subroutines.
+
+It is important that the names of constants follow a consistent and easily
+recognizable pattern.
+This way, the unique properties of constants are immediately communicated to
+the reader when observing any usage of a constant.
+For instance, they can be assured that the constant will not change, which
+simplifies debugging.
```
Comment (Swalkyn) [inline rules/S115/common/why-is-this-an-issue.adoc:11]: I would also use a simpler phrasing here, what do you think of:

```suggestion
This way, readers immediately understand that the referenced value does not change.
```

**Review by Swalkyn — CHANGES_REQUESTED** [review]
Looks good overall, but I think some of the explanations are not using the correct tone. I left some suggestions.

File: rules/S115/common/why-is-this-an-issue.adoc (hunk 1 above)
Comment (anton-haubner-sonarsource) [reply rules/S115/common/why-is-this-an-issue.adoc:11]: Well you are right, I was overcomplicating things here.
I like your suggestion, but I think we should also explain why this is useful (debugging), so I will try to keep that aspect, while also simplifying it.

File: rules/S115/common/what-is-the-potential-impact.adoc [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Applying a consistent naming scheme to constants improves the readability of
+code and makes it easier to understand, which will also boost team
+collaboration.
+Conversely, to forego a naming convention will negatively impact the overall
+maintainability of an application.
```
[the remark this answers is not in the document]
Comment (anton-haubner-sonarsource) [reply rules/S115/common/what-is-the-potential-impact.adoc:5]: Agreed, the sentences are too complex, and I like your simplification.
I would still like to retain the aspect that there is an impact on teamwork, so I'll try to fit that in while keeping it brief.

**Review by Swalkyn — APPROVED** [review]

**Pull request merged**